src/openms/source/ANALYSIS/OPENSWATH/MRMScoring.cpp

```diff
@@ -873,39 +897,52 @@ namespace OpenSwath
     {
       OPENSWATH_PRECONDITION(mi_precursor_matrix_.rows() > 1, "Expect mutual information matrix of at least 2x2");
 
+      const auto& em = mi_precursor_matrix_.getEigenMatrix();
+      size_t n_entries = em.size();
       double mi_scores{0};
-      for (const auto& e : mi_precursor_matrix_)
+
+      for (size_t i = 0; i < n_entries; i++)
       {
+        auto e = *(em.data() + i);
         mi_scores += e;
       }
+
       //mi_precursor_matrix_ is a triangular matrix
       size_t element_number = mi_precursor_matrix_.rows()*mi_precursor_matrix_.rows()/2 + (mi_precursor_matrix_.rows()+1)/2;
-      return mi_scores / element_number;
+      return mi_scores / (double)element_number;
     }
 
     double MRMScoring::calcMIPrecursorContrastScore()
     {
       OPENSWATH_PRECONDITION(mi_precursor_contrast_matrix_.rows() > 0 && mi_precursor_contrast_matrix_.cols() > 1, "Expect mutual information matrix of at least 1x2");
 
       double mi_scores{0};
-      for (const auto& e : mi_precursor_contrast_matrix_)
+      const auto& em = mi_precursor_contrast_matrix_.getEigenMatrix();
+      size_t n_entries = em.size();
+
+      for (size_t i = 0; i != n_entries; ++i)
```

Comment:
I would prefer "<" in index based for loops. What if someone starts to add stuff to i in the for loop.